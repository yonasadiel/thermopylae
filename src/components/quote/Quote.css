.quote {
    text-align: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 100%;
}

.quote .text-line {
    max-width: 500px;
    margin-bottom: .4em;
}

.quote .text-line span {
    font-size: 30px;
    white-space: pre-line;
}

.quote .subtext-line {
    /* position: absolute;
    left: 0;
    right: 0; */
    text-align: justify;
    max-width: 600px;
    margin-bottom: .4em;
}

.quote .subtext-line span {
    /* position: absolute; */
    right: 20px;
    white-space: pre-line;
}

.quote .nav-line {
    font-size: 8px;
}

.quote-settings .preloaded-option {
    margin-bottom: 0.2em;
}

.quote-settings .preloaded-option span {
    margin-left: 2px;
}

.quote-settings .custom-row {
    border-left: 1px solid var(--foreground-color);
    padding: 2px 10px;
    margin-bottom: 10px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    background-color: rgba(200, 200, 200, 0.1);
}

.quote-settings .custom-row input {
    font-size: 1.5em;
}

.quote-settings .custom-row button.delete {
    background-color: rgb(190, 0, 0);
    border-width: 0;
    margin-left: 12px;
}

.quote-settings .custom-row textarea {
    outline: none;
    background: transparent;
    color: white;
    border-width: 0 0 2px 0;
    padding: 5px 0;
    font-family: 'Raleway', sans-serif;
}

.quote-settings button {
    border-width: 0;
    font-family: unset;
    font-size: unset;
    color: unset;
    background-color: unset;
    cursor: pointer;
    color: var(--background-color);
    background-color: var(--foreground-color);
    border-radius: 4px;
    padding: 4px 8px;

}
