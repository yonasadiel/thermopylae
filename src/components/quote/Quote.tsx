import { useEffect, useState } from 'react';
import { filterPreloadedQuotes } from './util';
import { Quote } from '../../models/quote';
import { useSettings } from '../../hooks';
import preloadedQuotes from './preloaded';
import './Quote.css'

const ENABLE_NEXT_BUTTON = false;

const Quote = () => {
    const [activeQuotes, setActiveQuotes] = useState<Quote[]>([]);
    const { settings } = useSettings();
    useEffect(() => {
        setActiveQuotes([
            ...filterPreloadedQuotes(preloadedQuotes, settings.quotePreloaded),
            ...settings.quoteCustom,
        ]);
    }, [settings.quotePreloaded, settings.quoteCustom]);

    const seed = Math.floor((new Date()).getTime() / (60 * 60 * 1000));
    const quoteToShow = activeQuotes[seed % activeQuotes.length] || {};

    return (
        <div className="quote" style={{ display: settings.quoteEnabled ? 'flex' : 'none'}}>
            <div className="text-line">
                <span>{quoteToShow.text}</span>
            </div>
            <div className="subtext-line">
                <span>{quoteToShow.subtext}</span>
            </div>
            {ENABLE_NEXT_BUTTON && (
                <div className="nav-line">Next&gt;&gt;</div>
            )}
        </div>
    );
}

export default Quote;
