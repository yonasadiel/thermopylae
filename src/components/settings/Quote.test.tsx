import { test, expect, describe } from '@jest/globals';
import { render } from '@testing-library/react';
import Quote from './Quote';

describe('Quote Settings', () => {
    test('should render correctly', () => {
        const result = render(<Quote />);
        expect(result.baseElement).toMatchSnapshot();
    });
});