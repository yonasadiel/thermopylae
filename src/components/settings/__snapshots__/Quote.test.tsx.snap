// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Settings should render correctly 1`] = `
<body>
  <div>
    <div
      class="quote-settings"
    >
      <div
        class="input-group"
      >
        <label>
          General
        </label>
        <p>
          <input
            checked=""
            type="checkbox"
          />
          <span>
            Enabled
          </span>
        </p>
      </div>
      <div
        class="input-group"
      >
        <label>
          Preloaded Quotes
        </label>
        <p
          class="preloaded-option"
        >
          <input
            type="checkbox"
          />
          <span>
            Life Quotes
          </span>
        </p>
        <p
          class="preloaded-option"
        >
          <input
            checked=""
            type="checkbox"
          />
          <span>
            Pragmatic Programmer book
          </span>
        </p>
        <p
          class="preloaded-option"
        >
          <input
            checked=""
            type="checkbox"
          />
          <span>
            Programming
          </span>
        </p>
      </div>
      <div
        class="input-group"
      >
        <label>
          Custom Quotes
        </label>
        <p>
          <button
            type="button"
          >
            add
          </button>
        </p>
      </div>
    </div>
  </div>
</body>
`;
